public/store/DetectorsStore.tsx

```diff
@@ -0,0 +1,329 @@
+   */
+  readonly savedObjectsService: ISavedObjectsService;
+
+  /**
+   * Router history
+   * @property {RouteComponentProps['history']}
+   * @readonly
+   */
+  history: RouteComponentProps['history'] | undefined = undefined;
+
+  /**
+   * Keeps detector's data cached
+   *
+   * @property {IDetectorsCache} cache
+   */
+  private cache: IDetectorsCache = {};
+
+  /**
+   * Store state
+   * @private {IDetectorsState}
+   */
+  private state: IDetectorsState | undefined;
+
+  /**
+   * List of all shown toasts
+   * @private
+   */
+  private toasts: Toast[] = [];
+
+  constructor(
+    service: DetectorsService,
+    notifications: NotificationsStart,
+    savedObjectsService: ISavedObjectsService
+  ) {
+    this.service = service;
+    this.notifications = notifications;
+    this.savedObjectsService = savedObjectsService;
+  }
+
+  /**
+   * Invalidates all detectors data
+   */
+  private invalidateCache = (): DetectorsStore => {
+    this.cache = {};
+    return this;
+  };
+
+  public setState = (state: IDetectorsState, history: RouteComponentProps['history']): void => {
+    this.state = state;
+    this.history = history;
+
+    this.showNotification('Attempting to create the detector.', undefined, 'primary', true);
+  };
+
+  public getState = (): IDetectorsState | undefined => (this.state ? this.state : undefined);
+
+  public deleteState = (): void => {
+    delete this.state;
+  };
+
+  private showNotification = (
+    title: string,
+    message?: string,
+    type?: TCalloutColor,
+    loading?: boolean,
+    btnText?: string,
+    btnHandler?: (e: any) => void
+  ): void => {
+    if (!type) type = 'primary';
+
+    const closeAllToasts = () => {
+      this.toasts = [];
+      this.showToast(this.toasts);
+    };
+
+    const btn = btnText && (
+      <EuiButton
+        color={type}
+        onClick={(e: any) => {
+          btnHandler && btnHandler(e);
+          this.hideCallout();
+          closeAllToasts();
+        }}
+        size="s"
+      >
+        {btnText}
+      </EuiButton>
+    );
+
+    const messageBody = (
+      <EuiFlexGroup direction={'column'} alignItems={'flexStart'}>
+        {message && <EuiFlexItem grow={false}>{message}</EuiFlexItem>}
+        <EuiFlexItem grow={false}>{btn}</EuiFlexItem>
+      </EuiFlexGroup>
+    );
+
+    this.showCallout({
+      type,
+      title,
+      message: messageBody,
+      closeHandler: this.hideCallout,
+    });
+
+    const { color, iconType } = resolveType(type);
+    this.toasts.push({
+      title,
+      color,
+      iconType,
+      id: `toastsKey_${uuidv4()}`,
+      text: messageBody,
+    });
+    this.showToast(this.toasts);
+  };
+
+  private viewDetectorConfiguration = (): void => {
+    const pendingState = DataStore.detectors.getState();
+    const detectorState = pendingState?.detectorState;
+    this.history?.push({
+      pathname: `${ROUTES.DETECTORS_CREATE}`,
+      state: { detectorState },
+    });
+    DataStore.detectors.deleteState();
```

Comment:
Doubt: Does this delete happen even after the history push moves to different page?

Reply:
it does happen, but I did refactor it simply not to draw attention